demo/pkg/injector/context.go

```diff
@@ -0,0 +1,33 @@
+package injector
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/99designs/gqlgen/graphql/handler/transport"
+)
+
+type contextKey string
+
+const (
+	headerKey  contextKey = "header"
+	payloadKey contextKey = "payload"
+)
+
+func NewContextWithHeader(ctx context.Context, hdr http.Header) context.Context {
+	return context.WithValue(ctx, headerKey, hdr)
+}
+
+func NewContextWithInitPayload(ctx context.Context, initPayload transport.InitPayload) context.Context {
+	return context.WithValue(ctx, payloadKey, initPayload)
+}
+
+func Header(ctx context.Context) http.Header {
```

Comment:
Used at all?

Reply:
It's used here https://github.com/wundergraph/cosmo/blob/dfa1687064a3404547d4cab9f422a2bb0c795984/demo/pkg/subgraphs/test1/subgraph/schema.resolvers.go#L20